indra/belief/__init__.py

```diff
@@ -205,6 +209,86 @@ def check_prior_probs(self, statements):
                     raise Exception(msg)
 
 
+class BayesianScorer(SimpleScorer):
+    """This is a belief scorer which assumes a Beta prior and a set of prior
+    counts of correct and incorrect instances for a given source. It exposes
+    and interface to take additional counts and update its probability
+    parameters which can then be used to calculate beliefs on a set of
+    Statements.
+
+    Parameters
+    ----------
+    prior_counts : dict
+        A dictionary of counts of the form [pos, neg] for
+        each source.
+    subtype_counts : dict
+        A dictionary of counts of the form [pos, neg] for
+        each subtype within a source.
+    """
+    def __init__(self, prior_counts, subtype_counts):
+        self.prior_probs = load_default_probs()
+        self.subtype_probs = {}
+        self.prior_counts = prior_counts
+        self.subtype_counts = subtype_counts
+        # Set the probability estimates based on the counts
+        self.update_probs()
+
+    def update_probs(self):
+        """Update the internal probability values given the counts."""
+        # We deal with the prior probsfirst
+        # This is a fixed assumed value for systematic error
+        syst_error = 0.05
+        prior_probs = {'syst': {}, 'rand': {}}
+        for source, (p, n) in self.prior_counts.items():
+            # Skip if there are no actual counts
+            if n + p == 0:
+                continue
+            prior_probs['syst'][source] = syst_error
+            prior_probs['rand'][source] = \
+                1 - min((p / (n + p), 1-syst_error)) - syst_error
+        # Next we deal with subtype probs based on counts
+        subtype_probs = {}
+        for source, entry in self.subtype_counts.items():
+            for rule, (p, n) in entry.items():
+                # Skip if there are no actual counts
+                if n + p == 0:
+                    continue
+                if source not in subtype_probs:
+                    subtype_probs[source] = {}
+                subtype_probs[source][rule] = \
+                    1 - min((p / (n + p), 1-syst_error)) - syst_error
+        # Finally we propagate this into the full probability
+        # data structures of the parent class
+        super().update_probs(prior_probs, subtype_probs)
```

Comment:
Watch out for Python3 only usage of `super()` :)

Reply:
Yes, I knew I had to go back and fix that before merging :)